clippy_lints/src/methods/slice_as_bytes.rs

```diff
@@ -0,0 +1,39 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet_with_applicability, ty::is_type_lang_item};
+use rustc_errors::Applicability;
+use rustc_hir::{
+    Expr, ExprKind,
+    LangItem::{self, Range, RangeFrom, RangeFull, RangeTo, RangeToInclusive},
+    QPath,
+};
+use rustc_lint::LateContext;
+
+use super::SLICE_AS_BYTES;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>) {
+    if let ExprKind::Index(indexed, index) = recv.kind {
+        if let ExprKind::Struct(QPath::LangItem(Range | RangeFrom | RangeTo | RangeToInclusive | RangeFull, ..), ..) =
```

Comment:
There`s also `RangeInclusiveStruct` and `RangeInclusiveNew` not sure if they're relevant or not

Reply:
[From the documentation](https://doc.rust-lang.org/beta/nightly-rustc/rustc_hir/lang_items/enum.LangItem.html#variant.RangeInclusiveStruct)
> ### `RangeInclusiveStruct`
> The `RangeInclusive` lang item.

It should be added imo (Although, I don't think it will change anything, as `RangeInclusiveStruct` [isn't really used](https://github.com/search?q=repo%3Arust-lang%2Frust%20RangeInclusiveStruct&type=code))

But `RangeInclusiveNew` is the LangItem used for the `impl<idx> RangeInclusive<idx>` for creating a [new RangeInclusive](https://github.com/rust-lang/rust/blob/405130538938e2d63a761b7d8d6b27d26e634c33/library/core/src/ops/range.rs#L369), it shouldn't be added.

Reply:
There's https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/fn.is_range_literal.html which would also catch `..=`